lib/workers/repository/errors-warnings.ts

```diff
@@ -46,14 +50,60 @@ export function getDepWarnings(
               if (!warnings.includes(message)) {
                 warnings.push(message);
               }
-              if (!warningFiles.includes(file.packageFile)) {
+              if (
+                file.packageFile &&
+                !warningFiles.includes(file.packageFile)
+              ) {
                 warningFiles.push(file.packageFile);
               }
             }
           }
         }
       }
     }
+  } catch (err) {
+    // istanbul ignore next
+    logger.error({ err }, 'Error generating packageFiles');
+  }
+  return { warnings, warningFiles };
+}
+
+export function getDepWarningsDashboard(
+  packageFiles: Record<string, PackageFile[]>
+): string {
+  const { warnings, warningFiles } = getDepWarnings(packageFiles);
+  let warningText = '';
+  try {
+    if (!warnings.length) {
+      return '';
+    }
+    logger.debug(
+      { warnings, warningFiles },
+      'Found package lookup warnings in onboarding'
+    );
+    warningText = emojify(
+      `\n---\n\n### :warning: Dependency Lookup Warnings :warning:\n\n`
+    );
+    warnings.forEach((w) => {
```

Comment:
After a few prs that I've opened. I think we prefer a `for of` instead of a `forEach` loop

Reply:
mentioned in the best practices [Iterating objects & containers](https://github.com/MaronHatoum/renovate/blob/main/docs/development/best-practices.md#iterating-objects--containers)